layouts/partials/openapi/render-object-table.html

```diff
@@ -93,6 +94,29 @@
 
     {{ end }}
 
+   {{/* NB: the OpenAPI spec says "Consistent with JSON Schema, additionalProperties
+         defaults to true." Our schemas tend to omit additionalProperties; I think we
+         effectively assume it defaults to false. Therefore: we only output regarding
+         additionalProperties if it is explicitly set (i.e. is non-nil). */}}
+   {{ if (ne $additionalProperties nil )}}
+  <tr class="row-additional-properties">
+   <td colspan="3">
+        {{ if (and (reflect.IsMap $additionalProperties)) -}}
+            {{ if (index $additionalProperties "description") -}}
+May have additional properties: {{ $additionalProperties.description }}
+            {{ end }}
+        {{/* TODO: should we handle the case where additional properties are permitted,
+                    but must follow an explicit schema? */}}
+        {{ else if $additionalProperties -}}
+May have additional properties.
+        {{ else if (not $additionalProperties) -}}
+        {{/* At present we don't have any schema with additionalProperties: false.
+        But if we ever do, let's support it. */}}
+May **not** have additional properties.
```

Comment:
Everything is extensible in Matrix unless otherwise noted, so this is a bit misleading.

Reply:
I don't follow---what do you find misleading?

Reply:
Okay. Then perhaps this can be changed so that 

- we don't do anything if `additionalProperties: true`
- we raise a build-time error if `additionalProperties: false`
- Rather than "May have additional properties:" in the generated HTML, we write "Notes on additional properties:".

How does that sound?

Reply:
See https://swagger.io/specification/#schema-object

> The following properties are taken from the JSON Schema definition but their definitions were adjusted to the OpenAPI Specification.
> additionalProperties - Value can be boolean or object. Inline or referenced schema MUST be of a [Schema Object](https://swagger.io/specification/#schema-object) and not a standard JSON Schema. Consistent with JSON Schema, additionalProperties defaults to true.

Reply:
And for completeness: the original JSON schema additionalProperties are defined [here](https://json-schema.org/draft/2020-12/json-schema-core.html#additionalProperties).

Reply:
3398cf0f is an attempt at this.

Reply:
@turt2live, @DMRobertson: I'm not sure I'm following the discussion here. Can I just check that everyone is clear what we're discussing?

The question is around the correct behaviour if we encounter an object schema where `additionalProperties` is explictly set to `false` (as opposed to leaving it unset, which is equivalent to setting it to `true`).

[Earlier drafts](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.18) of JSON Schema give a clear definition of what that means:

> If "additionalProperties" is false, validation succeeds only if the instance is an object and all properties on the instance were covered by "properties" and/or "patternProperties".

In other words: this is explicitly about trying to define an object which **may not be extended**.

I don't entirely *object* to the decision to emit an error message, if we think it's somehow un-matrixy to try to define an object that isn't allowed any additional properties.  I'd just like to be sure that we made that decision with good information.

Reply:
We actually have that case (exactly one, unless I missed some weirdly formatted), and it's in [`m.receipt` events](https://github.com/matrix-org/matrix-spec/blob/a45138c3b4a722ef4f4eb98788dc25c94a3dce61/data/event-schemas/schema/m.receipt.yaml#L58): . `additionalProperties: false` is used next to `patternProperties` to forbid anything that does not follow the pattern. Which is quite valid and smart, in my opinion. There's a big question however on what the homeserver is supposed to do with this boundary. I would treat it as a "MUST ignore" of sorts, whereas omitting `additionalProperties` entirely would correspond to "MAY ignore". Does it make sense?